packages/mui-docs/src/i18n/i18n.tsx

```diff
@@ -107,8 +115,7 @@ export function useTranslate() {
           const fullKey = `${userLanguage}:${key}`;
           // No warnings in CI env
           if (!ignoreWarning && !warnedOnce[fullKey] && typeof window !== 'undefined') {
```

Comment:
Having conditions split between here and the `warn` function is odd and error prone. Maybe we could move all conditions to `warn` so we have all conditions centralized? And we just call `warn(fullKey)` here, same as in https://github.com/mui/material-ui/pull/42880/files. Also, the `!warnedOnce[fullKey]` condition is duplicated here and in the `warn` function.

Reply:
fair, refactored here -> https://github.com/mui/material-ui/pull/42881/commits/a7a1d2fcf7a4c54ea81e511be17d2900726ebcc3